js/parser.js

```diff
@@ -1,32 +1,244 @@
 /*
- * Copyright 2016 University of Southern California
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
+* Copyright 2016 University of Southern California
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
 
 var ERMrest = (function(module) {
 
     /**
-     * This is an internal function that parses a URI and constructs an
-     * internal representation of the URI.
-     * @memberof ERMrest
-     * @function _parse
-     * @private
-     */
-    module._parse = function(uri) {
-        // TODO parse a uri into some internal structure
+    * This is an internal function that parses a URI and constructs an
+    * internal representation of the URI.
+    * @memberof ERMrest
+    * @function _parse
+    * @param {String} uri - uri to be parsed
+    * @private
+    */
+    module._parse = function(uri, location) {
```

Comment:
`_parse` should not take the `location` argument. only the `uri`. this needs to be a fully qualified url to the ermrest resource. i.e., `https://example.org/ermrest/catalog/2/entity/s:t/k=123`. That is the only acceptable input to the `_parse` function. Therefore we do not need the `location` parameter. You can head with that in the chaise parse function _before_ you pass the ermrest url to the `resolve` function.